torch/csrc/autograd/custom_function.cpp

```diff
@@ -0,0 +1,84 @@
+#include <torch/csrc/autograd/custom_function.h>
+#include <torch/csrc/autograd/functions/accumulate_grad.h>
+
+namespace torch { namespace autograd {
+
+std::vector<Variable> _wrap_outputs(const std::unordered_set<at::TensorImpl*> &inputs,
```

Comment:
A useful C++ tip to know for the future: `unordered_map` and `unordered_set` are actually really crappy hash map implementations and perform pretty badly, and there is usually something faster in reach. We have one in `c10/util/flat_hash_map.h`. I don't know if it matters here, since this code isn't that fast anyway, but if you ever are in a situation where you need a hash map, there you go.

Reply:
(And the old code used `unordered_set` anyway, so, it's no worse!)